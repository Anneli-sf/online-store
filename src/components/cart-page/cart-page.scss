.body {
    background-color: #7f0909;
}
.cart-wrapper {
    position: relative;
    width: 100vw;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: self-start;
    padding-bottom: 100px;
    input {
        background-color: transparent;
    }
    .cart-products {
        background-attachment: fixed;
        width: 60%;
        flex: 0 1 60%;
        margin-right: 15px;
        height: 70vh;
    }

    .block-title {
        font-size: 28px;
    }

    .cart-products {
        .block-header {
            width: 100%;
            display: flex;
            align-items: center;
            color: #ffc500;
            padding: 10px 20px;
            margin-bottom: 10px;
            justify-content: space-between;
            background-color: transparent;
        }
        .quantity,
        .current-page {
            background-color: transparent;
            color: #ffc500;
            pointer-events: none;
            text-align: center;
            font-size: 20px;
        }

        .label,
        .btns-container-switch-pages {
            display: flex;
            align-items: center;
        }

        .title-of-container-switch-btns {
            margin-right: 10px;
        }

        .description-content {
            padding: 10px;
        }

        .btn-switch-page-left,
        .btn-switch-page-right {
            position: relative;
            width: 20px;
            height: 20px;
            background-color: transparent;
            border: none;
            &:hover {
                transform: scale(0.9);
            }
        }

        .btn-switch-page-left::after {
            position: absolute;
            content: url('../../public/icons/arrow-down.svg');
            top: -20px;
            right: -35px;
            transform: rotate(90deg);
        }

        .btn-switch-page-right::after {
            position: absolute;
            content: url('../../public/icons/arrow-down.svg');
            top: -15px;
            right: 20px;
            transform: rotate(-90deg);
            margin-left: 10px;
        }
        .pagination {
            border: 1px solid #ffc500;
            color: #ffc500;
            padding: 5px;
            width: 100px;
            &::placeholder {
                color: #ffc500;
            }
        }
    }

    .stock-value {
        cursor: auto;
        width: 40px;
        text-align: end;
        font-size: 20px;
    }
    .quantity-items-cart {
        text-align: center;
        width: 50px;
        font-size: 20px;
    }

    .cart-products .block-section {
        padding: 10px;
        height: 88%;
        overflow: auto;
        &::-webkit-scrollbar {
            width: 0px;
        }

        .block-header,
        .description-header {
            padding: 5px 0px;
            position: relative;
            background-color: transparent;
            color: #7f0909;
            &::after {
                position: absolute;
                content: '';
                bottom: 0px;
                width: 105%;
                height: 1px;
                background-color: #7f0909;
            }
        }
        .cart-list {
            width: 100%;
            display: flex;
            flex-direction: column;
        }

        .cart-item {
            background-color: #ffc500;
            display: flex;
            justify-content: space-between;
            padding: 5px;
            width: 100%;
            color: #7f0909;
            border-radius: 5px;
            &:not(:last-child) {
                margin-bottom: 10px;
            }
        }

        .cart-item-image {
            width: 130px;
            border-radius: 10px;
        }

        .number-product {
            flex: 1 0 2%;
            margin-right: 10px;
            display: flex;
            justify-content: center;
            align-items: center;
            color: #7f0909;
        }

        .cart-item-image {
            flex: 0 1 20%;
            margin-right: 10px;
            cursor: pointer;
        }

        .description-cart-item {
            margin-right: 10px;
            width: 100%;
            overflow: hidden;
            cursor: pointer;
        }

        .mini-description-item {
            white-space: nowrap;
            text-overflow: ellipsis;
            overflow: hidden;
            max-width: 70%;
            margin: 0 auto;
            margin-bottom: 10px;
        }

        .block-title {
            text-align: center;
            color: #7f0909;
        }

        .values-cart-item {
            position: relative;
            display: flex;
            flex-direction: column;
            justify-content: space-around;
            align-items: center;
            padding: 10px;
            width: 40%;
        }

        .raiting-discount-container {
            display: flex;
            justify-content: space-around;
        }

        .raiting,
        .discount {
            text-decoration: underline;
        }

        .add-del-btns-container {
            display: flex;
            justify-content: space-between;
            align-items: center;
        }

        .add-item {
            background-color: transparent;
            border: none;
            position: relative;
            width: 20px;
            height: 20px;
            &:hover {
                transform: scale(0.9);
            }
            &::after {
                position: absolute;
                content: url('../../public/icons/plus.svg');
                width: 20px;
                top: -2px;
                left: -15px;
            }
        }

        .delete-item {
            background-color: transparent;
            border: none;
            position: relative;
            width: 20px;
            height: 20px;
            &:hover {
                transform: scale(0.9);
            }
            &::before {
                position: absolute;
                content: url('../../public/icons/minus.svg');
                width: 20px;
                top: -2px;
                left: 0px;
            }
        }
        .container {
            width: 25%;
        }
        .total-price {
            position: relative;
            font-size: 20px;
            width: 50px;
            text-align: left;
        }
        .span {
            position: absolute;
            right: 0px;
            top: 0px;
            font-size: 20px;
        }
    }

    .summary {
        width: 25%;
        align-self: center;
        border: 1px solid #ffc500;
        border-radius: 5px;

        .block-header {
            width: 100%;
            display: flex;
            align-items: center;
            padding: 10px 20px;
            margin-bottom: 10px;
            color: #ffc500;
        }
        .block-section {
            display: flex;
            flex-direction: column;
            justify-content: space-between;
            align-items: center;
            text-align: center;

            input {
                text-align: center;
                background-color: transparent;
                color: #ffc500;
            }
        }
        .block-section {
            display: flex;
            justify-content: space-between;
            align-items: center;
        }

        .quantity-products,
        .total-sum,
        .promecode-input,
        .promo-test {
            color: #ffc500;
            margin-bottom: 10px;
        }

        .promecode-input {
            border: 1px solid #ffc500;
            border-radius: 5px;
        }

        .total-sum-value {
            position: relative;
            font-size: 20px;
        }

        .quantity-products-value {
            font-size: 20px;
        }

        .promecode-input {
            background-color: transparent;
            padding: 5px 20px;
            color: #ffc500;
            &:focus {
                -webkit-box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2) inset;
                -moz-box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2) inset;
                box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2) inset;
            }
            &::placeholder {
                color: #ffc500;
            }
        }

        .form-buy {
            padding: 10px;
        }

        .promo-test {
            font-size: 16px;
            opacity: 0.1;
            &:hover {
                opacity: 1;
            }
        }

        .btn-buy-now {
            background-color: transparent;
            border: 1px solid #ffc500;
            border-radius: 5px;
            color: #ffc500;
            padding: 5px 20px;
            &:hover {
                -webkit-box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2) inset;
                -moz-box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2) inset;
                box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2) inset;
            }
        }
    }
}
