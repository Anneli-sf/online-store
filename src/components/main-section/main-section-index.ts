import { IProductsData, IFilters } from '../global-components/interfaces';
import { productsData } from '../data/data';
import { deleteDoubleAddUnique, addDoubleDeleteUnique } from '../helpers/helpers';
import { openPopup, closePopup } from './products-section/products-section';
import { deleteCheckBoxStyles } from './aside/aside';

//------------------main function for checkbox filters
export const findCurrentFilters = (el: HTMLInputElement, filters: IFilters, minPrice: string, maxPrice: string) => {
    const chosenCategory: IProductsData[] = productsData.filter((item) => item.categoryEng === el.getAttribute('id'));
    const chosenSubCategory: IProductsData[] = productsData.filter(
        (item) => item.subcategoryEng === el.getAttribute('id')
    );

    filters.categories = deleteDoubleAddUnique(filters.categories, chosenCategory);
    // .filter(
    //     (item) => item.price >= +minPrice && item.price <= +maxPrice
    // );
    // console.log('filters.categories', filters.categories);

    //-----------содержит ли категории выбранные подкатегории
    if (filters.categories.length > 0) {
        let isContain = 0;
        filters.categories.forEach((item) => {
            chosenSubCategory.forEach((element) => {
                if (item.categoryEng == element.categoryEng) isContain++;
            });
        });
        // console.log('isContain', isContain);

        if (isContain == 0 && chosenSubCategory.length > 0) {
            filters.subcategories = filters.subcategories;
            // .filter((item) => item.price >= +minPrice && item.price <= +maxPrice);
            el.checked === true ? showNotFound() : closePopup();
        } else {
            filters.subcategories = deleteDoubleAddUnique(filters.subcategories, chosenSubCategory);
            // .filter((item) => item.price >= +minPrice && item.price <= +maxPrice);
            closePopup();
        } //---------------если категория не выбрана
    } else {
        filters.subcategories = deleteDoubleAddUnique(filters.subcategories, chosenSubCategory);
        // .filter((item) => item.price >= +minPrice && item.price <= +maxPrice);
    }

    // console.log('filters.subcategories', filters.subcategories);

    filters.currArr =
        filters.subcategories.length === 0
            ? filters.categories
            : filters.categories.length === 0
            ? filters.subcategories
            : addDoubleDeleteUnique(filters.categories, filters.subcategories);

    //----------------RESET FILTERS
    if (el.classList.contains('input-reset')) {
        filters.currArr = [];
    }
    // console.log('max, min', maxPrice, minPrice);
    if (filters.currArr.length === 0) {
        filters.currArr = productsData;
        filters.categories = [];
        filters.subcategories = [];
        deleteCheckBoxStyles();
        return filters.currArr;
    } else return filters.currArr; //.filter((item) => item.price >= +minPrice && item.price <= +maxPrice);
    // return filters.currArr.length === 0 ? productsData : filters.currArr;
};

//---------------------create URL
export const stateFilters = (
    categories: string[],
    subcategories: string[],
    minPrice: string,
    maxPrice: string,
    minAmount: string,
    maxAmount: string,
    resultArr: IProductsData[],
    // element?: HTMLInputElement
) => {
    let categoryState = '';
    let subcategoryState = '';
    // let searchWord = '';

    if (categories.length > 0) categoryState = 'category=' + categories.join('↕');
    if (subcategories.length > 0) subcategoryState = 'subcategory=' + subcategories.join('↕');
    // if (element.className === 'sort__input' && element.value.length > 0) searchWord = `&search=${element.value}`;
    let state = `?${categoryState}&${subcategoryState}&price=${minPrice}↕${maxPrice}&stock=${minAmount}↕${maxAmount}${searchWord}`;
    if (resultArr.length === productsData.length) state = '/';

    return state;
};

//-------------------find cards bu searching
export const searchByWord = (word: string, arr: IProductsData[]) => {
    word = word.toLowerCase();
    if (word.length !== 0)
        return arr.filter((item) => {
            if (
                item.category.toLowerCase().includes(word) ||
                item.subcategory.toLowerCase().includes(word) ||
                item.title.toLowerCase().includes(word) ||
                item.description.toLowerCase().includes(word) ||
                item.price.toString().includes(word) ||
                item.discount.toString().includes(word) ||
                item.stock.toString().includes(word) ||
                item.raiting.toString().includes(word)
            )
                return item;
        });
    else return arr;
};

//---------------show goods are not founded
export const showNotFound = (): void => {
    const productsList = document.querySelector('.products__list') as HTMLUListElement;
    const sortSpan = document.querySelector('.found-items') as HTMLSpanElement;
    productsList.style.display = 'none';
    openPopup();
    sortSpan.innerHTML = '0';
};

//--------------------------set prices to slider
export const setPricesToSlider = (resultArr: IProductsData[]): void => {
    const minPrice = document.querySelector('#min-price') as HTMLInputElement;
    const maxPrice = document.querySelector('#max-price') as HTMLInputElement;
    const minPriceNumber = document.querySelector('#min-input-price') as HTMLInputElement;
    const maxPriceNumber = document.querySelector('#max-input-price') as HTMLInputElement;
    const prices: number[] = resultArr.reduce((total: Array<number>, item: IProductsData) => {
        total.push(item.price);
        return total;
    }, []);

    minPrice.value = Math.min(...prices).toString();
    maxPrice.value = Math.max(...prices).toString();
    minPriceNumber.value = minPrice.value;
    maxPriceNumber.value = maxPrice.value;
};

//--------------------------set amount to slider
export const setAmountToSlider = (resultArr: IProductsData[]): void => {
    const minAmount = document.querySelector('#min-amount') as HTMLInputElement;
    const maxAmount = document.querySelector('#max-amount') as HTMLInputElement;
    const minAmountNumber = document.querySelector('#min-input-amount') as HTMLInputElement;
    const maxAmountNumber = document.querySelector('#max-input-amount') as HTMLInputElement;
    const stock: number[] = resultArr.reduce((total: Array<number>, item: IProductsData) => {
        total.push(item.stock);
        return total;
    }, []);

    minAmount.value = Math.min(...stock).toString();
    maxAmount.value = Math.max(...stock).toString();
    minAmountNumber.value = minAmount.value;
    maxAmountNumber.value = maxAmount.value;
};
