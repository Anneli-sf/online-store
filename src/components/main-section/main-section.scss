.products-wrapper {
    width: 100%;
    height: 85vh;
    overflow: auto;
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    padding-top: 10px;
    &::-webkit-scrollbar {
        width: 0px;
    }
}