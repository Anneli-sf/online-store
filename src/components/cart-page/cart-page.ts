import './cart-page.scss';
import {
    createElement,
    createBlock,
    createButton,
    createLabel,
    createImage,
    createInput,
    createSimpleInput,
    createParagraph,
} from '../global-components/global-components';
import { productsData } from '../data/data';

const listBlock = createElement('ul', 'cart-list') as HTMLUListElement;

export const createCartPage = (): HTMLDivElement => {
    const cartPage = createElement('div', 'cart-wrapper') as HTMLDivElement;
    cartPage.append(createProductsCartBlock(), createSummaryCartBlock());
    return cartPage;
};

function sliceIntoChunks(arr: Array<number>, chunkSize: number) {
    const res = [];
    for (let i = 0; i < arr.length; i += chunkSize) {
        const chunk = arr.slice(i, i + chunkSize);
        res.push(chunk);
    }
    return res;
}

export const createProductsCartBlock = () => {
    const cartBlock = createElement('div', 'cart-products');
    const cartBlockHeader = createElement('div', 'block-header');
    const cartBlockTitle = createElement('p', 'block-title');
    cartBlockTitle.textContent = 'Товары в корзине';

    const cartBlockContent = createElement('div', 'block-section');

    cartBlockHeader.append(cartBlockTitle, quantityProductsOnCart(), switchPagesBlock());

    const fillCartPages = () => {
        const arr = JSON.parse(localStorage.getItem('cartList') as string);
        if (arr.length === 0) {
            return sliceIntoChunks(JSON.parse(localStorage.getItem('cartList') as string), 3);
        } else {
            return sliceIntoChunks(JSON.parse(localStorage.getItem('cartList') as string), 3)[
                +localStorage.getItem('currentPage') - 1
            ];
        }
    };

    const idArray = fillCartPages();
    listBlock.innerHTML = '';
    idArray.forEach((item) => {
        createProductsList(item);
    });

    cartBlockContent.append(listBlock);

    cartBlock.querySelectorAll('.number-product').forEach((item, index) => {
        item.textContent = `${index + 1}`;
    });

    cartBlock.append(cartBlockHeader, cartBlockContent);
    // document.querySelector('.main')?.append(cartBlock);
    return cartBlock;
};

export const createProductsList = (productId: number): HTMLUListElement => {
    listBlock.append(cartProductBlock(productId));

    return listBlock;
};
//----------------------------HEADER
const quantityProductsOnCart = () => {
    const labelQuantityOfProducts = createLabel('Товары: ', 'quantity');
    const inputQuantityOfProducts = createSimpleInput(
        'quantity',
        'number',
        '',
        `${JSON.parse(localStorage.getItem('cartList') as string).length}`
    ) as HTMLInputElement;
    labelQuantityOfProducts.append(inputQuantityOfProducts);

    return labelQuantityOfProducts;
};

if (!localStorage.getItem('currentPage')) {
    localStorage.setItem('currentPage', '1');
}

const switchPagesBlock = () => {
    const pagesSwitchesButtonsBlock = createElement('div', 'btns-container-switch-pages') as HTMLDivElement;

    const currentPage = createSimpleInput(
        'current-page',
        'number',
        '',
        `${localStorage.getItem('currentPage')}`
    ) as HTMLInputElement;

    const textPage = createParagraph('Страница: ', 'title-of-container-switch-btns') as HTMLParagraphElement;

    const buttonSwitchPagesToLeft = createButton('', 'btn-switch-page-left');
    const buttonSwitchPagesToRight = createButton('', 'btn-switch-page-right');

    if (localStorage.getItem('btnLeft') === 'show') {
        buttonSwitchPagesToLeft.style.transform = 'scale(1)';
    } else if (localStorage.getItem('btnLeft') === 'hide') {
        buttonSwitchPagesToLeft.style.transform = 'scale(0)';
    }

    if (localStorage.getItem('btnRight') === 'show') {
        buttonSwitchPagesToRight.style.transform = 'scale(1)';
    } else if (localStorage.getItem('btnRight') === 'hide') {
        buttonSwitchPagesToRight.style.transform = 'scale(0)';
    }

    pagesSwitchesButtonsBlock.append(textPage, buttonSwitchPagesToLeft, currentPage, buttonSwitchPagesToRight);

    return pagesSwitchesButtonsBlock;
};

//----------------------------/HEADER

const cartProductBlock = (productId?: number) => {
    const productCartBlock = createElement('li', 'cart-item') as HTMLLIElement;
    productCartBlock.id = `${productId}`;
    productCartBlock.append(
        numberOfProductBlock(productId),
        imageProductBlock(productId),
        productsDescriptionBlock(productId),
        productsValuesBlock(productId)
    );

    return productCartBlock;
};

const numberOfProductBlock = (productId: number) => {
    const numberOfProduct = createElement('p', 'number-product') as HTMLParagraphElement;
    numberOfProduct.textContent = `${JSON.parse(localStorage.getItem('cartList') as string).indexOf(productId) + 1}`;
    return numberOfProduct;
};

const imageProductBlock = (productId: number): HTMLImageElement => {
    const image = createImage(
        productsData[productId].images[0],
        productsData[productId].title,
        'cart-item-image'
    ) as HTMLImageElement;
    return image;
};

const productsDescriptionBlock = (productId: number) => {
    const descriptionBlock = createElement('div', 'description-cart-item') as HTMLDivElement;
    const descriptionHeader = createElement('div', 'description-header') as HTMLDivElement;
    const descriptionTitle = createParagraph(productsData[productId].title, 'block-title') as HTMLParagraphElement;

    const descriptionContent = createElement('div', 'description-content') as HTMLDivElement;

    const miniDescription = createParagraph(
        productsData[productId].description,
        'mini-description-item'
    ) as HTMLParagraphElement;

    const raitingDiscountContainer = createElement('div', 'raiting-discount-container') as HTMLDivElement;

    const raiting = createParagraph('Рейтинг: ', 'raiting') as HTMLParagraphElement;
    const discount = createParagraph('Скидка: ', 'discount') as HTMLParagraphElement;

    const raitingValue = createSimpleInput('raiting-value', '', '', `${productsData[7].raiting}`) as HTMLInputElement;
    raitingValue.setAttribute('readonly', 'true');

    const discountValue = createSimpleInput(
        'discount-value',
        '',
        '',
        `${productsData[7].discount}%`
    ) as HTMLInputElement;
    discountValue.setAttribute('readonly', 'true');

    raiting.append(raitingValue);
    discount.append(discountValue);
    raitingDiscountContainer.append(raiting, discount);

    descriptionHeader.append(descriptionTitle);
    descriptionContent.append(miniDescription, raitingDiscountContainer);
    descriptionBlock.append(descriptionHeader, descriptionContent);

    return descriptionBlock;
};

const productsValuesBlock = (productId: number) => {
    const valuesBlock = createElement('div', 'values-cart-item') as HTMLDivElement;
    const stock = createParagraph('Доступно: ', 'stock') as HTMLParagraphElement;

    if (!localStorage.getItem(`stock_${productId}`)) {
        localStorage.setItem(`stock_${productId}`, productsData[productId].stock - 1);
    }

    const stockValue = createSimpleInput(
        'stock-value',
        'number',
        '',
        `${localStorage.getItem(`stock_${productId}`)}`,
        '',
        '',
        '',
        `${productId}`
    ) as HTMLInputElement;
    stockValue.setAttribute('readonly', 'true');
    const shtuk = createElement('span', 'stock-text') as HTMLDivElement;
    shtuk.textContent = `шт.`;

    if (!localStorage.getItem(`quantityProduct_${productId}`)) {
        localStorage.setItem(`quantityProduct_${productId}`, '1');
    }

    const quantityOfItemsInCart = createSimpleInput(
        'quantity-items-cart',
        'number',
        '',
        +localStorage.getItem(`quantityProduct_${productId}`)
    ) as HTMLInputElement;
    quantityOfItemsInCart.setAttribute('readonly', 'true');

    const addAndDelItemsButtonsContainer = createElement('div', 'add-del-btns-container') as HTMLDivElement;
    const addItemButton = createButton('', 'add-item') as HTMLButtonElement;
    addItemButton.id = `${productId}`;
    const deleteItemButton = createButton('', 'delete-item') as HTMLButtonElement;
    deleteItemButton.id = `${productId}`;

    const container = createElement('div', 'container') as HTMLDivElement;
    const totalPrice = createElement('input', 'total-price') as HTMLInputElement;
    totalPrice.readOnly = true;
    totalPrice.id = `${productId}`;
    if (!localStorage.getItem(`price_${productId}`)) {
        localStorage.setItem(`price_${productId}`, `${productsData[productId].price}`);
        totalPrice.value = `${productsData[productId].price}`;
    } else {
        (totalPrice.value as string | null) = localStorage.getItem(`price_${productId}`);
    }
    const span = createElement('span', 'span');
    span.textContent = '$';
    container.append(totalPrice, span);

    stock.append(stockValue, shtuk);
    addAndDelItemsButtonsContainer.append(deleteItemButton, quantityOfItemsInCart, addItemButton);
    valuesBlock.append(stock, addAndDelItemsButtonsContainer, container);

    return valuesBlock;
};

//-----------------------------------------------------

//------------------STOCK and QUANTITY------------------
export const createSummaryCartBlock = () => {
    const summaryBlock = createBlock('summary', 'Итого') as HTMLElement;

    const summarySectionBlock = summaryBlock.querySelector('.block-section') as HTMLElement;

    const quantityOfPoducts = createParagraph('Товары, шт: ', 'quantity-products') as HTMLParagraphElement;

    let totalStock = JSON.parse(localStorage.getItem('cartList') as string).length;

    if (localStorage.getItem('totalStock')) {
        totalStock = +localStorage.getItem('totalStock');
    }

    const quantityOfPoductsValue = createSimpleInput(
        'quantity-products-value',
        '',
        '',
        `${totalStock}`,
        '',
        '',
        true
    ) as HTMLInputElement;
    const totalSum = createParagraph('Сумма: ', 'total-sum') as HTMLParagraphElement;

    let totalPrice = JSON.parse(localStorage.getItem('cartItems') as string)
        .flat()
        .reduce((acc, curr) => acc + curr.price, 0);

    if (localStorage.getItem('totalPrice')) {
        totalPrice = +localStorage.getItem('totalPrice');
    }
    const totalSumValue = createSimpleInput(
        'total-sum-value',
        '',
        '',
        `${totalPrice}`,
        '',
        '',
        true
    ) as HTMLInputElement;

    summarySectionBlock.append(
        quantityOfPoducts,
        quantityOfPoductsValue,
        totalSum,
        totalSumValue,
        formOfBuyProductsBlock()
    );
    // document.querySelector('.main')?.append(summaryBlock);

    return summaryBlock;
};

const formOfBuyProductsBlock = () => {
    const buyForm = createInput('promecode-input', 'text', 'Введите промокод') as HTMLFormElement; //'form-buy'

    const promoTest = createParagraph("Попробуйте: 'Гарри', 'Поттер'", 'promo-test') as HTMLParagraphElement;
    const buttonBuyNow = createButton('Купить сейчас', 'btn-buy-now');

    buyForm.append(promoTest, buttonBuyNow);

    return buyForm;
};
