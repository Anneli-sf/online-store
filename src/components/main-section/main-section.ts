import './main-section.scss';
import { createElement } from '../global-components/global-components';
import { createAside } from './aside/aside';
import { createProductsSection } from './products-section/products-section';
import { IProductsData } from '../data/data';

export const productsWrapper = createElement('div', 'products-wrapper') as HTMLDivElement;
export function createProducstPage(currentArr: IProductsData[]): HTMLDivElement {
    // const productsWrapper = createElement('div', 'products-wrapper') as HTMLDivElement;
    productsWrapper.append(createAside(currentArr), createProductsSection(currentArr));//вернула

    return productsWrapper;
}
