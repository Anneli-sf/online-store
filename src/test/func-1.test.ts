import { createEmptyPage, sliceIntoChunks } from '../components/cart-page/cart-page';
import { productsData } from '../components/data/data';
import { stateFilters } from '../components/main-section/main-section';

test('If block created?', () => {
    expect(createEmptyPage()).toMatchSnapshot();
});

test('Is the array divided into parts correctly?', () => {
    expect(sliceIntoChunks([1, 2, 3, 4], 2)).toEqual(
        expect.arrayContaining([
            [1, 2],
            [3, 4],
        ])
    );
});

test('Does slash mean home page?', () => {
    expect(stateFilters([], [], productsData)).toBe('/');
});
