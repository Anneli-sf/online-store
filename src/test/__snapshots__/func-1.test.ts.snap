// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If block created? 1`] = `
<div
  class="empty"
>
  <p
    class="empty-text"
  >
    Корзина пуста...
  </p>
</div>
`;
